src/app/data-client.ts

```diff
@@ -687,6 +687,44 @@ export class DataClient {
 
     return filter;
   }
+
+  /**
+   * Gets the most recent tabular data captured from the specified data source,
+   * as long as it was synced within the last year.
+   *
+   * @param partId The ID of the part that owns the data
+   * @param resourceName The name of the requested resource that captured the
+   *   data
+   * @param resourceSubtype The subtype of the requested resource that captured
+   *   the data
+   * @param methodName The data capture method name
+   * @returns A tuple containing [timeCaptured, timeSynced, payload] or null if
+   *   no data has been synced for the specified resource OR the most recently
+   *   captured data was over a year ago
+   */
+  async getLatestTabularData(
+    partId: string,
+    resourceName: string,
+    resourceSubtype: string,
```

Comment:
How would y'all feel about swapping these 2? I kinda like the idea of progressive specificity -- go from least specific (part) to most specific (method name). 

So the order would be part -> subtype -> name -> method

Reply:
I think if you guys don't mind, we'd like to keep the order as is, at least in this PR - this follows the existing ordering convention for data that's shown in the UI, among other places. We're already exposing this ordering to users (below is a screenshot from the image / sensor view's filter panel) so would like to keep things consistent.

<img width="245" alt="Screenshot 2024-12-03 at 9 53 18 AM" src="https://github.com/user-attachments/assets/44eb6836-f657-4956-8526-5b5dd3337e10">

Of course, happy to discuss further on this - the field order switch could definitely be a larger change across data